runtime/drivers/duckdb/olap.go

```diff
@@ -181,6 +186,344 @@ func (c *connection) EstimateSize() (int64, bool) {
 	return fileSize(paths), true
 }
 
+// AddTableColumn implements drivers.OLAPStore.
+func (c *connection) AddTableColumn(ctx context.Context, tableName, columnName, typ string) error {
+	if !c.config.TableAsView {
+		return c.Exec(ctx, &drivers.Statement{
+			Query:    fmt.Sprintf("ALTER TABLE %s ADD COLUMN %s %s", safeSQLName(tableName), safeSQLName(columnName), typ),
+			Priority: 1,
+		})
+	}
+
+	version, exist, err := c.tableVersion(tableName)
+	if err != nil {
+		return err
+	}
+
+	if !exist {
+		return fmt.Errorf("table %q does not exist", tableName)
+	}
+	dbName := dbName(tableName, version)
+	// ignore query cancellations
+	ctx = context.WithValue(context.Background(), connCtxKey{}, connFromContext(ctx))
+	err = c.Exec(ctx, &drivers.Statement{
+		Query:    fmt.Sprintf("ALTER TABLE %s.default ADD COLUMN %s %s", safeSQLName(dbName), safeSQLName(columnName), typ),
+		Priority: 1,
+	})
+	if err != nil {
+		return err
+	}
+
+	// recreate view to propagate schema changes
+	return c.Exec(ctx, &drivers.Statement{
+		Query:    fmt.Sprintf("CREATE OR REPLACE VIEW %s AS SELECT * FROM %s.default", safeSQLName(tableName), safeSQLName(dbName)),
```

Comment:
Why does it need to re-create the view if only adding a column? It would still be doing `SELECT * FROM` the same underlying table.

Reply:
This puzzled me as well but it seems view is no longer queryable if the schema of the underlying table changes.
![image](https://github.com/rilldata/rill/assets/12948312/e21a2523-37e7-4127-a4e3-4acd8b33ee2b)

Reply:
Weird...